cpp/src/arrow/buffer.h

```diff
@@ -137,6 +139,25 @@ class ARROW_EXPORT Buffer {
   /// \return a new Buffer instance
   static std::shared_ptr<Buffer> FromString(std::string data);
 
+  /// \brief Construct an immutable buffer that takes ownership of the contents
+  /// of an std::vector (without copying it).
+  ///
+  /// \param[in] data a string to own
+  /// \return a new Buffer instance
+  template <typename T>
+  static std::shared_ptr<Buffer> FromVector(std::vector<T> vec) {
+    auto* data = reinterpret_cast<uint8_t*>(vec.data());
+    auto size = static_cast<int64_t>(vec.size());
+    return std::shared_ptr<Buffer>(
+        new Buffer{data, size},
+        [vec = std::move(
+             vec)  // Keep the vector's buffer alive inside the destructor until after we
+                   // have deleted the Buffer. Note we can't use this trick in FromString
+                   // since std::string's data is inline for short strings so moving
+                   // invalidates pointers into the string's buffer.
```

Comment:
`clang-format` will have a nicer time if you place this comment after the `,` above.

Reply:
Or just use `// clang-format off` / `// clang-format on` around those lines.